File: src/com/mistraltech/smogen/utils/ActionUtils.java
package com.mistraltech.smogen.utils;

import com.intellij.openapi.application.ApplicationManager;
import com.intellij.openapi.command.CommandProcessor;
import com.intellij.openapi.project.Project;

public class ActionUtils {
    public static void runAction(Runnable action) {
        ApplicationManager.getApplication().runWriteAction(action);
    }

    public static void runActionAsCommand(final Runnable action, final Project project) {
        CommandProcessor.getInstance().executeCommand(project, () -> runAction(action), "", "");
    }

    public static void runActionLater(final Runnable action) {
        ApplicationManager.getApplication().invokeLater(() -> runAction(action));
    }

}

File: src/main/kotlin/com/mistraltech/smogen/utils/ActionUtils.kt
package com.mistraltech.smogen.utils

import com.intellij.openapi.application.ApplicationManager
import com.intellij.openapi.command.CommandProcessor
import com.intellij.openapi.project.Project

object ActionUtils {
    @JvmStatic
    fun runAction(action: Runnable?) {
        ApplicationManager.getApplication().runWriteAction(action!!)
    }

    fun runActionAsCommand(action: Runnable?, project: Project?) {
        CommandProcessor.getInstance().executeCommand(project, { runAction(action) }, "", "")
    }

    fun runActionLater(action: Runnable?) {
        ApplicationManager.getApplication().invokeLater { runAction(action) }
    }
}
